### PR TITLE
[zero-copy] stabilize InputRef

### PR DESCRIPTION
**Pull request by CraftSpider**
Makes public the minimal API for custom parsers - no state access, non non-terminal errors, but you can parse, recover, and error correctly.

### CODE REVIEW

File: src/zero_copy/input.rs [hunk 1]
```diff
@@ -222,7 +222,7 @@ impl<'a, 'parse, I: Input + ?Sized, E: Error<I>, S> InputRef<'a, 'parse, I, E, S
         }
     }
 
-    pub(crate) fn next(&mut self) -> (usize, Option<I::Token>) {
+    pub fn next(&mut self) -> (usize, Option<I::Token>) {
```
Comment (zesterer) [inline src/zero_copy/input.rs:225]: Part of me wonders whether we should stabilise the offset being returned. Perhaps we could have a `next_token` input that's just a light wrapper around `next` but drops the first field? What do you think?

Comment (CraftSpider) [reply src/zero_copy/input.rs:225]: Currently the `usize` is used for `Located::at`, however, if we changed it to `Marker` we could drop the first field. I like the idea, it further reduces the chance of issues with user-facing API lock-in

Comment (zesterer) [reply src/zero_copy/input.rs:225]: Sure! That seems like a sensible solution.

**Comment by zesterer** [comment]
Ready to merge?

**Comment by CraftSpider** [comment]
Yep

**Pull request merged**